File: 백준/Silver/1978. 소수 찾기/소수 찾기.java

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
	public static void main(String[] args) throws IOException {
		int max = 1001;
		boolean[] isPrime = new boolean[max];

		isPrime[0] = isPrime[1] = true;
		
		for(int i=2;i<=Math.sqrt(max-1);i++) {
			if(isPrime[i] == true) {
				continue;
			}

			for(int j= i*2; j < max; j = j + i) {
				isPrime[j] = true;
			}
		}
		
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int num = Integer.parseInt(br.readLine());
		int[] numlist = new int[num];
		
		StringTokenizer st = new StringTokenizer(br.readLine());
		int primeCnt = 0;
		for(int i=0; i<num; i++) {
			if(isPrime[Integer.parseInt(st.nextToken())]) {
				continue;
			} else primeCnt++;
		}
		
		System.out.println(primeCnt);
	}
}

File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    int max = 1000001;
    
    vector<bool> isPrime(max, true);
    isPrime[0] = false;
    isPrime[1] = false;
    
    for (int i=2; i*i<max; i++) {
        if (!isPrime[i]) continue;
        
        for (int j=i*i; j<max; j+=i) {
            isPrime[j] = false;
        }
    }
    int answer = 0;

    for (int i=1; i<=n; i++) {
        if (isPrime[i]) answer++;
    }
    
    return answer;
}